### PR TITLE
ui-speedspacechart: enhance tooltip

### LINKED ISSUE #553: ui-speedspacechart: enhance tooltip
### Description and goal

some UX/UI enhancements for GEV tooltip

### Acceptance criteria

- [x] don't show the speed if it equals 0
- [x] numbers in IBM plex Mono
- [x] display "--" if an info is missing
- [x] disable the display animation, always display the tooltip

### PR DESCRIPTION
**Pull request by Yohh**
close #553

### CODE REVIEW

**Comment by Caracol3** [comment]
It appears that sometimes the electrical profiles are displayed below the line voltage and are misaligned compared to the rest.
image

File: ui-speedspacechart/src/components/common/DetailsBox.tsx [hunk 1]
```diff
@@ -43,19 +43,19 @@ const DetailsBox = ({
...
-  const adaptedOffset = store.isSettingsPanelOpened ? 641 : 115;
+  const detailsBoxWidth = document.getElementById('details-box')?.offsetWidth || 0;
+  const detailsBoxHeight = document.getElementById('details-box')?.offsetHeight || 0;
 
   // find out if the box is going out off the right side of the canvas
-  if (curveX + adaptedOffset > width - MARGIN_RIGHT - 10) rightOffset = 127;
+  if (curveX + detailsBoxWidth > width - MARGIN_RIGHT - 10) rightOffset = detailsBoxWidth + 12;
   // find out if the box is going out off the bottom side of the canvas
-  if (curveY + 180 > height - MARGIN_BOTTOM - 10) bottomOffset = 192;
+  if (curveY + detailsBoxHeight > height - MARGIN_BOTTOM - 10) bottomOffset = detailsBoxHeight + 12;
```
Comment (flomonster) [inline ui-speedspacechart/src/components/common/DetailsBox.tsx]: Can we avoid having this magic number?

**Review by flomonster — APPROVED** [review]
LGTM.

> [!NOTE]
> A code refactor idea, we should not provide speeds as text but numbers. Currently, the provided speed is serialized and then deserialized.

File: ui-speedspacechart/src/components/common/DetailsBox.tsx (hunk 1 above)
Comment (Yohh) [reply ui-speedspacechart/src/components/common/DetailsBox.tsx]: ~~I think you wasn't in the right branch, it's already wrote this way~~
edit: you're talking about the `10`number?

Comment (flomonster) [reply ui-speedspacechart/src/components/common/DetailsBox.tsx]: I know this file already has a lot of magic numbers. We should rewrite this in another PR.
I was talking of the `12` which is new.

Comment (Yohh) [reply ui-speedspacechart/src/components/common/DetailsBox.tsx]: np it's done

**Review by Caracol3 — APPROVED** [review]
lgtm (tested)

**Review by Math-R — APPROVED** [review]
lgtm :)

**Review by SarahBellaha — APPROVED** [review]
💯 lgtm

**Pull request merged**